sunpy/data/_sample.py

```diff
@@ -129,27 +129,27 @@ def get_sample_file(filename, url_list, show_progress=True, overwrite=False,
                 url = urljoin(base_url, online_filename)
                 exists = url_exists(url)
                 if exists:
-                    f = download_file(os.path.join(base_url, online_filename),
+                    f = download_file(str(Path(base_url).joinpath(online_filename)),
```

Comment:
I'm not sure whether `Path` is what we want to use for online stuff.. Path's are not urls. In fact, why there's a `url` above, and then when called for `download_file` is joint using `os.path.join`?

Reply:
If `download_file` returns a `Path`, then the code below becomes cleaner.

Reply:
should i remove Paths from functions involving internet access?

Reply:
It depends on what that path is being used for. If it's for the download location, then it should be fine. If it's somehow for a URL, it needs to removed.